lib/utilities/sign.js

```diff
@@ -0,0 +1,52 @@
+// This file is meant to work with node.js
+// TODO: handshake compatibility
+const { MTX } = require('bcoin');
+const { LedgerTXInput } = require('bledger');
+const assert = require('bsert');
+
+/*
+ * @param {Object} hwApp
+ * @param {Object} ptx - proposal transaction
+ * @param {String} account - bip44 account
+ * @param {String} enc - return signature in hex
+ */
+async function hardwareSignatures(hwApp, ptx, account, enc) {
+  const { tx, scripts, paths } = ptx;
+  assert(tx, 'requires proposal tx txn');
+  assert(scripts, 'requires proposal tx scripts');
+  assert(paths, 'requires proposal tx paths');
+
+  const inputCount = tx.inputs.length;
+  assert(inputCount === scripts.length, 'must have one script for each input');
+  assert(inputCount === paths.length, 'must have one path for each input');
+
+  const mtx = MTX.fromJSON(tx);
+  const ledgerInputs = [];
+
+  for (const [i, input] of mtx.inputs.entries()) {
+    const { branch, index } = paths[i];
+    const path = `${account}/${branch}/${index}`;
+    const redeem = Buffer.from(scripts[i]);
+    const coin = mtx.view.getCoinFor(input);
+    const ledgerInput = new LedgerTXInput({
+      witness: true,
```

Comment:
Not 100% sure if this should be hardcoded to `true`

Reply:
This can be read from the tx right? My feeling is that if we can avoid hardcoding it in we should, but could default to `true` as a fallback if/when necessary.